pkgs/by-name/yt/ytmdesktop/package.nix

```
@@ -0,0 +1,97 @@
+{
+  lib,
+  asar,
+  binutils,
+  commandLineArgs ? "",
+  electron_30,
+  fetchurl,
+  makeDesktopItem,
+  makeWrapper,
+  nix-update-script,
+  stdenvNoCC,
+  zstd,
+}:
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "ytmdesktop";
+  version = "2.0.5";
+
+  meta = {
+    changelog = "${finalAttrs.meta.downloadPage}/tag/v${finalAttrs.version}";
+    description = "A Desktop App for YouTube Music";
+    downloadPage = "https://github.com/ytmdesktop/ytmdesktop/releases";
+    homepage = "https://ytmdesktop.app/";
+    license = lib.licenses.gpl3Only;
+    mainProgram = finalAttrs.pname;
+    maintainers = [ lib.maintainers.cjshearer ];
+    inherit (electron_30.meta) platforms;
+    # While the files we extract from the .deb are cross-platform (javascript), the installation
+    # process for darwin is different, and I don't have a test device. PRs are welcome if you can
+    # add the correct installation steps. I would suggest looking at the following:
+    # https://www.electronjs.org/docs/latest/tutorial/application-distribution#manual-packaging
+    # https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/instant-messengers/jitsi-meet-electron/default.nix
+    badPlatforms = lib.platforms.darwin;
+  };
+
+  passthru.updateScript = nix-update-script { };
+
+  desktopItem = makeDesktopItem {
+    desktopName = "Youtube Music Desktop App";
+    exec = finalAttrs.pname;
+    icon = finalAttrs.pname;
+    name = finalAttrs.pname;
+    genericName = finalAttrs.meta.description;
+    mimeTypes = [ "x-scheme-handler/ytmd" ];
+    categories = [
+      "AudioVideo"
+      "Audio"
+    ];
+    startupNotify = true;
+  };
+
+  nativeBuildInputs = [
+    asar
+    binutils
+    electron_30
+    makeWrapper
+    zstd
+  ];
+
+  src = fetchurl {
+    url = "${finalAttrs.meta.downloadPage}/download/v${finalAttrs.version}/youtube-music-desktop-app_${finalAttrs.version}_amd64.deb";
+    hash = "sha256-0j8HVmkFyTk/Jpq9dfQXFxd2jnLwzfEiqCgRHuc5g9o=";
+  };
+
+  unpackPhase = ''
+    ar x $src data.tar.zst
+    tar xf data.tar.zst
+  '';
+
+  patchPhase = ''
+    pushd usr/lib/youtube-music-desktop-app
+
+    asar extract resources/app.asar patched-asar
+
+    # workaround for https://github.com/electron/electron/issues/31121
+    sed -i "s#process\.resourcesPath#'$out/lib/resources'#g" \
+      patched-asar/.webpack/main/index.js
+
+    asar pack patched-asar resources/app.asar
+
+    popd
+  '';
+
+  installPhase = ''
+    mkdir -p $out/{lib,share/pixmaps}
+
+    cp -r usr/lib/youtube-music-desktop-app/{locales,resources{,.pak}} $out/lib
+    cp usr/share/pixmaps/youtube-music-desktop-app.png $out/share/pixmaps/${finalAttrs.pname}.png
```

```suggestion
    cp usr/share/pixmaps/youtube-music-desktop-app.png $out/share/pixmaps/ytmdesktop.png
```
that will be in any other case most likely wrong